src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> optionalUser = userService.findByEmail(email);
+        if (!email.isEmpty() && !password.isEmpty() && optionalUser.isPresent()) {
+            User user = optionalUser.get();
+            String currentlyPassword = HashUtil.hashPassword(password, user.getSalt());
+            if (user.getPassword().equals(currentlyPassword)) {
+                return user;
+            }
+        }
```

Comment:
```suggestion
        if (!email.isEmpty() && !password.isEmpty() && optionalUser.isPresent() && checkPassword(optionalUser.get(), password)) {
            return optionalUser.get();
        }
```
let's create a separate private method `checkPassword(User user, String password)`